healthcheck/healthcheck_test.go

```diff
@@ -136,60 +137,67 @@ func TestSetBackendsConfiguration(t *testing.T) {
 		},
 	}
 
-	for _, test := range tests {
-		test := test
-		t.Run(test.desc, func(t *testing.T) {
-			t.Parallel()
-			// The context is passed to the health check and canonically cancelled by
-			// the test server once all expected requests have been received.
-			ctx, cancel := context.WithCancel(context.Background())
-			defer cancel()
-			ts := newTestServer(cancel, test.healthSequence)
-			defer ts.Close()
-
-			lb := &testLoadBalancer{RWMutex: &sync.RWMutex{}}
-			backend := NewBackendHealthCheck(Options{
-				Path:     "/path",
-				Interval: healthCheckInterval,
-				LB:       lb,
+	for _, withPort := range []bool{false, true} {
```

Comment:
I don't think we need to test all permutations here. I suggest we write a test for the newly created `newRequest` method instead where we won't need to jump through as many hoops (as we do in this test).